src/main/java/pl/tomaszdziurko/jvm_bloggers/blogs/BloggersDataUpdater.java

```diff
@@ -104,8 +104,7 @@ protected boolean somethingChangedInBloggerData(Blog blog, BloggerEntry bloggerE
 
     private boolean urlFromRssIsValidAndDifferentThanExistingOne(Blog blog,
```

Comment:
I think the method should be refactored and split into two separate methods. One for validity checking and the second one for equality checking. This way we can clearly see if this blog's url was blank or/and equal to existing one while investigating or debugging errors.